features/facebookImport/test/importers/ad-view-language-importer.test.js

```diff
@@ -14,140 +14,94 @@ const datasets = [
-        expect(thirdAd.viewedTimestamps).toStrictEqual([1693455618]);
-
-        //has correct view counts for ads
-        expect(firstAd.viewsCount).toBe(2);
-        expect(secondAd.viewsCount).toBe(1);
-        expect(thirdAd.viewsCount).toBe(1);
-    });
-});
-
-/*
-let testInputs = null;
-
-beforeAll(() => {
-    const datasets = [
-        { language: "English", dataset: createEnglishAdViewsData() },
-        { language: "German", dataset: createGermanAdViewsData() },
-    ];
-
-    return Promise.all(
-        datasets.map(async ({ language, dataset }) => {
-            const zipFile = new ZipFileMock();
-            zipFile.addJsonEntry(RECENTLY_VIEWED_FILE_PATH, dataset);
-            const importingResult = await runRecentlyViewedAdsImporter(zipFile);
-
-            return { language, importingResult };
-        })
-    ).then((importingResults) => {
-        testInputs = importingResults.map(({ language, importingResult }) => {
-            const result = importingResult.result;
-            const facebookAccount = importingResult.facebookAccount;
-            const relatedAccounts = facebookAccount.relatedAccounts;
-            const [firstRelatedAccount, secondRelatedAccount] =
-                relatedAccounts.items;
-            return [
-                language,
-                [
-                    result,
-                    facebookAccount,
-                    relatedAccounts,
-                    firstRelatedAccount,
-                    secondRelatedAccount,
-                ],
-            ];
+datasets.forEach(([language, dataset]) =>
+    describe(`Import ad views from ${language} dataset`, () => {
+        let result = null;
+        let relatedAccounts = null;
+        let firstRelatedAccount = null;
+        let secondRelatedAccount = null;
+
+        beforeAll(async () => {
+            const importingResult = await runAdsImportForDataset(dataset);
+            result = importingResult.result;
+            relatedAccounts = importingResult.relatedAccounts;
+            [firstRelatedAccount, secondRelatedAccount] = relatedAccounts.items;
         });
-    });
-});
-
-describe("Import ad views", () => {
-    console.log(testInputs);
-    test.each(testInputs)(
-        "returns success status for %s dataset.",
-        (language, result) => {
-            expectImportSuccess(result);
-        }
-    );
-
-    test.each(testInputs)(
-        "has two related accounts for %s dataset.",
-        (language, { relatedAccounts }) => {
-            expect(relatedAccounts.count).toBe(2);
-        }
-    );
-
-    test.each(testInputs)(
-        "has three ads for %s dataset.",
-        (language, { relatedAccounts }) => {
-            expect(relatedAccounts.adsCount).toBe(3);
-        }
-    );
-});
-*/
+
+        it("returns success status", () => expectImportSuccess(result));
+
+        it("has two related accounts", () =>
+            expect(relatedAccounts.count).toBe(2));
```

Comment:
Literals in tests are known to be brittle. Can you compute that from the original test data you're using?

Reply:
Added `AD_VIEWS_EXPECTED_DATA` and put the total values for multiple types of entries there.